django/db/migrations/serializer.py

```diff
@@ -155,11 +155,6 @@ def serialize(self):
             raise ValueError("Cannot serialize function: lambda")
         if self.value.__module__ is None:
             raise ValueError("Cannot serialize function %r: No module" % self.value)
-        # Python 3 is a lot easier, and only uses this branch if it's not local.
```

Comment:
In the second commit (along with some other cleanups), I readded a test similar to the one removed in https://github.com/django/django/commit/c716fe87821df00f9f03ecc761c914d1682591a2#diff-d073e708462e9844f28fdfa568043d51 which it seems should have always run on Python 3 (without the expectation of an exception).